src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java

```diff
@@ -37,61 +51,25 @@ public List<OrderDetails> getAllOrders() {
 
         List<OrderDetails> list = new ArrayList<>(orders.size());
         for (Order order : orders) {
-            OrderDetails orderDetails = collectOrderDetails(order);
-            list.add(orderDetails);
+            list.add(collectOrderDetails(order));
         }
         return list;
     }
 
-
-    private OrderDetails collectOrderDetails(Order order) {
-        OrderDetails orderDetails = new OrderDetails();
-        orderDetails.setId(order.getId());
+    private OrderDetails collectOrderDetails(Order order){
+        OrderDetails orderDetails = orderMapper.toOrderDetails(order);
         orderDetails.setOrderItems(collectOrderItemDetails(order.getId()));
-        orderDetails.setCustomerId(order.getUserId());
-        orderDetails.setUpdatedAt(order.getUpdatedAt());
-        orderDetails.setCreatedAt(order.getCreatedAt());
         orderDetails.setTotalPrice(getTotalPriceForOrder(orderDetails.getOrderItems()));
         return orderDetails;
     }
 
-    private BigDecimal getTotalPriceForOrder(List<OrderItemDetails> orderItems) {
-        BigDecimal result = BigDecimal.ZERO;
-        for (OrderItemDetails orderItem : orderItems) {
-            BigDecimal totalPrice = orderItem.getProduct().getPrice()
-                    .multiply(BigDecimal.valueOf(orderItem.getQuantity()));
-            result = result.add(totalPrice);
-        }
-        return result;
-    }
-
     private List<OrderItemDetails> collectOrderItemDetails(Integer orderId) {
-        return orderItemRepository.getOrderItemsByOrderId(orderId).stream()
-                .map(orderItem -> convertOrderItemToOrderItemDetails(orderItem))
-                .toList();
+        List<OrderItem> orderItemsByOrderId = orderItemRepository.getOrderItemsByOrderId(orderId);
+        List<Integer> productIds = orderItemsByOrderId.stream().map(OrderItem::getProductId).toList();
```

Comment:
what id different order items have the same product id? you will get duplicates in the `productIds`, right?

Reply:
yes you are right, we can duplicate in the productIds